api/src/services/graphql/utils/sanitize-gql-schema.ts

```diff
@@ -0,0 +1,101 @@
+import type { Relation, SchemaOverview } from '@directus/types';
+import { useLogger } from '../../../logger.js';
+
+/**
+ * Regex was taken from the spec
+ * https://spec.graphql.org/June2018/#sec-Names
+ */
+const GRAPHQL_NAME_REGEX = /^[_A-Za-z][_0-9A-Za-z]*$/;
+
+// Not an exhaustive list (doesnt include generated names)
+const GRAPHQL_RESERVED_NAMES = [
+	'Subscription',
+	'Query',
+	'Mutation',
+	'Int',
+	'Float',
+	'String',
+	'Boolean',
+	'DateTime',
+	'ID',
+	'uid',
+	'Point',
+	'PointList',
+	'Polygon',
+	'MultiPolygon',
+];
+
+/**
+ * Filters out invalid collections to prevent graphql from errorring on schema generation
+ *
+ * @param schema
+ * @returns
+ */
+export function sanitizeGraphqlSchema(schema: SchemaOverview) {
+	const logger = useLogger();
+
+	const collections = Object.entries(schema.collections).filter(([collectionName, _data]) => {
+		if (collectionName.startsWith('__') || !collectionName.match(GRAPHQL_NAME_REGEX)) {
```

Comment:
Let's clarify either via a comment or alternatively putting `__` into a const
```suggestion
		// '__' is reserved for GraphQL introspection system
		if (collectionName.startsWith('__') || !collectionName.match(GRAPHQL_NAME_REGEX)) {
```